analyses/malawi/scripts/mwi_chirps_dry_spell_viz.R

```diff
@@ -44,49 +31,210 @@ fill_dates <- function(df_dates, start_col, end_col, fill_num){
       }
     }
   }
-  
   # Clean up the df to a tidy format for ggplot
   df_long <- df %>%
     gather(pcode, day_type, head(adms,1):tail(adms,1))%>%
     mutate(day_fac = as.factor(day_type))%>%
     mutate(month_day = format(dates, "%m-%d"))%>%
     mutate(date_no_year = as.Date(paste0('1800-',month_day), "%Y-%m-%d"))%>%
     mutate(year = lubridate::year(dates))%>%
-    mutate(region = substr(pcode, 3, 3)) %>% 
+    mutate(region = substr(pcode, 3, 3)) %>%
+    mutate(region = ifelse(region == 3, "Southern", ifelse(region == 2, "Central", "Northern")))
+
+  return(df_long)
+}
+
+#part of function fill_dates, used if data already in semi-processed format
+prepare_ggplot <- function(df,ds_col){
+  df$dates <- as.Date(df$date,format="%Y-%m-%d")
+  df$day_type <- df[[ds_col]]
+  df_long <- df %>%
+    mutate(day_fac = as.factor(day_type))%>%
+    mutate(month_day = format(dates, "%m-%d"))%>%
+    mutate(date_no_year = as.Date(paste0('1800-',month_day), "%Y-%m-%d"))%>%
+    mutate(year = lubridate::year(dates)) %>%
+    mutate(region = substr(pcode, 3, 3)) %>%
     mutate(region = ifelse(region == 3, "Southern", ifelse(region == 2, "Central", "Northern")))
-  
   return(df_long)
 }
 
 
-df_ds <- fill_dates(df_dry_spells_mean, 'dry_spell_first_date', 'dry_spell_last_date', 1)
-df_rs <- fill_dates(df_rainy_season_mean, 'onset_date', 'cessation_date', 10)
+
+plot_heatmap <- function(df_dry_spells,df_rainy_season, match_values,match_labels,color_scale,y_label,plot_title,output_path,ds_flatdata=FALSE,sub_title=""){
+  #ds_flatdata: choose the appropriate preprocessing, depending on the format the data comes in
+  #the output from mwi_chirps_dry_spell_detection.R should use ds_flatdata=FALSE
+  #however, some cases the data is preprocessed in python in which case ds_flatdata=TRUE
+  if (ds_flatdata) {
+    df_ds <- prepare_ggplot(df_dry_spells,"dryspell_match")
+  }
+  else {
+      df_ds <- fill_dates(df_dry_spells, df_rainy_season,'dry_spell_first_date', 'dry_spell_last_date', 1)
+  }
+  df_rs <- fill_dates(df_rainy_season, df_rainy_season,'onset_date', 'cessation_date', 10)
+
+  df_ds %>%
```

Comment:
Not critical but could probably clean this up a bit by defining the custom theme separately: 

```
theme_hm <- theme_minimal()+
    theme(
      axis.text.y=element_blank(),
       axis.ticks.y=element_blank(),
       legend.position = 'bottom',
       strip.text = element_text(size=16,angle=0),
       axis.text.x = element_text(size=16),
       legend.text = element_text(size=16),
       axis.title.x = element_text(size=16),
       axis.title.y = element_text(size=16),
       plot.title = element_text(size=32)
    )
```